.github/workflows/pr-lint.yaml

```diff
@@ -35,3 +35,21 @@ jobs:
           bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash)
           ./actionlint -color -shellcheck=
         shell: bash
+
+  # Special lint to handle reusable workflows requiring an explicit branch.
+  # We cannot just reference the current/default but have to make sure it is `master`.
+  reusable-workflow_lint-pr:
+    runs-on: ubuntu-latest
+    name: PR - Reusable workflow linting
+    steps:
+      # We probably should target this on only workflow changes but the complexity of doing that is a lot more than just letting the grep run.
+      - uses: actions/checkout@v2
+      # Recursive grep or any workflow using a local reusable workflow but not ending in @master
+      - run: |
+          if [[ $(grep -R -E 'uses:[ ]*fluent/fluent-bit/.github/workflows/.*@[^master].*$' .github/workflows/*.y*ml| wc -l) -gt 0 ]]; then
+            echo "Found reusable workflow on non-master branch"
+            grep -R -E 'uses:[ ]*fluent/fluent-bit/.github/workflows/.*@[^master].*$' .github/workflows/*.y*ml
```

Comment:
maybe make the branch parametrized? or via a env?

Reply:
possibly but it might just get super-complex for something hopefully we can chuck away once Github supports it?